src/subdomains/lightning/services/lightning-currency.service.ts

```diff
@@ -1,15 +1,19 @@
 import { BadRequestException, Injectable, InternalServerErrorException, OnModuleInit } from '@nestjs/common';
 import { Currency } from '@uma-sdk/core';
+import { Config } from 'src/config/config';
 import { LightningLogger } from 'src/shared/services/lightning-logger';
 import { LightningHelper } from '../../../integration/blockchain/lightning/lightning-helper';
 import { CoinGeckoService } from '../../support/services/coingecko.service';
+import { LightingWalletPaymentParamDto } from '../dto/lightning-wallet.dto';
 
 @Injectable()
 export class LightningCurrencyService implements OnModuleInit {
   private readonly logger = new LightningLogger(LightningCurrencyService);
 
   private currencyCache: Map<string, Currency>;
 
+  private currencyFormat = /^([$₣€§])?(\d+\.?\d*)$/;
+

```

Comment:
Should we construct the regex programatically with the registered currencies?

Reply:
We can do this when we retrieve the currencies from the database. As we currently define the currencies statically in the program code, I think it is sufficient if we also define the symbols statically in the RegEx expression.

Reply:
Ok, then we shouldn't forget about it 😉 👍